ios/RNDateTimePickerShadowView.m

```diff
@@ -0,0 +1,49 @@
+#import "RNDateTimePickerShadowView.h"
+
+@implementation RNDateTimePickerShadowView
+
+- (instancetype)init
+{
+  if (self = [super init]) {
+    YGNodeSetMeasureFunc(self.yogaNode, RNDateTimePickerShadowViewMeasure);
+  }
+  return self;
+}
+
+- (void)setDate:(NSDate *)date {
+  _date = date;
+  YGNodeMarkDirty(self.yogaNode);
+}
+
+- (void)setLocale:(NSLocale *)locale {
+  _locale = locale;
+  YGNodeMarkDirty(self.yogaNode);
+}
+
+- (void)setMode:(UIDatePickerMode)mode {
+  _mode = mode;
+  YGNodeMarkDirty(self.yogaNode);
+}
+
+static YGSize RNDateTimePickerShadowViewMeasure(YGNodeRef node, float width, YGMeasureMode widthMode, float height, YGMeasureMode heightMode)
+{
+  RNDateTimePickerShadowView *shadowPickerView = (__bridge RNDateTimePickerShadowView *)YGNodeGetContext(node);
+
+  __block CGSize size;
+  dispatch_sync(dispatch_get_main_queue(), ^{
```

Comment:
We need to execute these calls on the main thread. This does have _some_ performance impact but should be negligible (might post some numbers later)